src/shell/file_reader.cpp

```diff
@@ -29,7 +29,7 @@ std::optional<FileReader> FileReader::GetFileReader(const std::string& filename)
 		return {};
 	}
 	DOS_CloseFile(handle);
-	return FileReader(std::move(fullname));
+	return std::unique_ptr<FileReader>(new FileReader(std::move(fullname)));
```

Comment:
Isn't `return new FileReader...` enough here?

Reply:
Doesn't compile.  `std::make_unique` doesn't work either because it's a private constructor so I think this is the best way.

```
../../src/shell/file_reader.cpp:32:9: error: no viable conversion from returned value of type 'FileReader *' to function return type 'std::unique_ptr<FileReader>'
        return new FileReader(std::move(fullname));
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Sure.